### PR TITLE
Improve layer information extraction

### PR DESCRIPTION
**Pull request by denizetkar**
# Description
- `SquadServer` regularly invokes `updateLayerInformation` to update the current and next layer variables, namely `this.currentLayer` and `this.nextLayer`.
- This is fetched from `Layers` class, `getLayerByName` method, which filters the pulled layers `this.layers` by the `name` field.
- The layers are fetched from `https://raw.githubusercontent.com/Squad-Wiki/squad-wiki-pipeline-map-data/master/completed_output/_Current%20Version/finished.json`
  - I'm actually a bit concerned that this "current version" is outdated with respect to the latest Squad version 8.0.
- However, the layer name for filtering is fetched from `SquadRcon` class, `getCurrentMap` method which uses the RCON command `ShowCurrentMap`. This command returns an answer like `Current level is Jensen's Range, layer is JensensRange_USA-RGF, factions USA RGF`. The problem here is 2-fold:
  - The regex used to extract the level and the layer names is broken: `^Current level is (.*), layer is (.*)` Because the 2nd regex group actually matches a string like `JensensRange_USA-RGF, factions USA RGF`
  - Even if we correctly extract the "layer name", it actually corresponds to the `layerid` field in the pulled layers

## What this PR does
Solves those 2 problems by:
- Updating the regexes to extract the level and the layer ids from the RCON commands `ShowCurrentMap` and `ShowNextMap` as: `^Current level is ([^,]*), layer is ([^,]*)`
- Changing the matching logic to use the `layerid` field instead of `name`

### CODE REVIEW

File: squad-server/plugins/discord-killfeed.js
Comment (Ulibos) [inline squad-server/plugins/discord-killfeed.js:1]: This seems to be an accidental edit but other than that looks good to me. Going to test it soon.

File: squad-server/plugins/discord-killfeed.js [hunk 1]
```diff
@@ -82,11 +82,6 @@ export default class DiscordKillFeed extends DiscordBasePlugin {
           ? `[${info.victim.steamID}](https://steamcommunity.com/profiles/${info.victim.steamID})`
           : 'Unknown',
         inline: true
-      },
-      {
```
Comment (werewolfboy13) [inline squad-server/plugins/discord-killfeed.js]: Why is this removed? Will need to be added back in if you want this PR approved.

**Review by werewolfboy13 — CHANGES_REQUESTED** [review]
Changes are required.

File: squad-server/plugins/discord-killfeed.js (hunk 1 above)
Comment (denizetkar) [reply squad-server/plugins/discord-killfeed.js]: To quote myself,

> In addition, removes a redundant code in `DiscordKillFeed` class that used to cause an access error, where if `info.victim` was `null` then the field access statement `info.victim.name` would fail, causing SquadJS to crash.

I observed this behavior randomly in my own dedicated server while the code on `master` branch was running. It would throw access error while trying to access `info.victim.name` when `info.victim` was `null`.

After removing this bit, I can confirm that I haven't encountered that error ever since. Also, if you carefully check that list variable, there is already a duplicate object:
```
      {
        name: "Victim's Name",
        value: info.victim ? info.victim.name : 'Unknown',
        inline: true
      },
```

I hope that was enough reason.

Comment (werewolfboy13) [reply squad-server/plugins/discord-killfeed.js]: That isn't a reason to pull it off. It needs to be corrected as there are users that needs to use that info. The crash will need to be corrected elsewhere in the code.

Comment (werewolfboy13) [reply squad-server/plugins/discord-killfeed.js]: Also each change needs to have its own PR. This change isn't tied at all to the main core issue/correction of this PR. That's why I accidentally overlooked it.

Comment (denizetkar) [reply squad-server/plugins/discord-killfeed.js]: done

Comment (denizetkar) [reply squad-server/plugins/discord-killfeed.js]: Although I still do not understand how that is not a duplicate object. The same field is literally defined 2 positions behind except with a value that doesn't fail in case `info.victim` is `null` 🤷 
```typescript
    const fields = [
      {
        name: "Attacker's Name",
        value: info.attacker.name,
        inline: true
      },
      {
        name: "Attacker's SteamID",
        value: `[${info.attacker.steamID}](https://steamcommunity.com/profiles/${info.attacker.steamID})`,
        inline: true
      },
      {
        name: "Attacker's EosID",
        value: info.attacker.eosID,
        inline: true
      },
      {
        name: 'Weapon',
        value: info.weapon
      },
      {
        name: "Victim's Name",
        value: info.victim ? info.victim.name : 'Unknown',
        inline: true
      },
      {
        name: "Victim's SteamID",
        value: info.victim
          ? `[${info.victim.steamID}](https://steamcommunity.com/profiles/${info.victim.steamID})`
          : 'Unknown',
        inline: true
      },
      {
        name: "Victim's Name",
        value: info.victim.name,
        inline: true
      }
    ];
```

Comment (Ulibos) [reply squad-server/plugins/discord-killfeed.js]: While your fix is meaningful in this particular case (I can only assume there was a "bad" merge that caused the duplication) werewolf asks to keep killfeed fix separate from this specific issue simply for clarity. I am sure that if you commit this fix as a separate PR (and providing the rationale from above) it will be merged in.

**Comment by Ulibos** [comment]
I have submitted a PR that addresses duplicate Victim field that you reported: #364

**Comment by denizetkar** [comment]
Hi @Ulibos,

Any updates on testing the PR?

**Comment by Ulibos** [comment]
We are using a plugin that queries data from layers infrequently so I can't vouch for a full comprehensive test but so far I haven't seen a single failure on this part of the code.

**Comment by denizetkar** [comment]
Should we merge it then?

**Comment by Ulibos** [comment]
I'm all for merge but I'm not the one calling the shots.

**Comment by werewolfboy13** [comment]
There is another merge about to happen soon that will require rewrites again. Merges are on hold until then.

**Review by werewolfboy13 — APPROVED** [review]

**Review by werewolfboy13 — APPROVED** [review]

**Comment by werewolfboy13** [comment]
Force pushing. Mobile UI is preventing conversation resolution.
